### PR TITLE
🌱 Fix lint issues: Replace golint with revive

### PR DESCRIPTION
**Pull request by chrismcgehee**
* **Please check if the PR fulfills these requirements**
- [ n/a] Tests for the changes have been added (for bug fixes / features)
- [x] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Code quality improvement

* **What is the current behavior?** (You can also link to an open issue here)
[golint](https://github.com/golang/lint) is now deprecated. [golangci-lint](https://golangci-lint.run/usage/linters/) recommends replacing it with [revive](https://github.com/mgechev/revive). This PR does that. Most of the changes revive wants are to add comments to exported variables and methods. \
Part of issue #173.

* **What is the new behavior (if this is a feature change)?**
n/a

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

### CODE REVIEW

File: cron/config/config.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 )
 
 const (
+	// ShardNumFilename is the name of the file that stores the shard number.
```
Comment (azeemshaikh38) [inline cron/config/config.go:29]: `stores the number of shards`

File: roundtripper/roundtripper.go [hunk 2]
```diff
@@ -27,15 +27,23 @@ import (
...
+	GithubAuthToken = "GITHUB_AUTH_TOKEN" // #nosec G101
+	// GithubAppKeyPath is the path to file for GitHub App key.
+	GithubAppKeyPath = "GITHUB_APP_KEY_PATH"
+	// GithubAppID is the app ID for the GitHub App.
+	GithubAppID = "GITHUB_APP_ID"
+	// GithubAppInstallationID is the installation ID for the GitHub App.
+	GithubAppInstallationID = "GITHUB_APP_INSTALLATION_ID"
+	// UseDiskCache will cache results to disk for subsequent runs if set to true.
+	UseDiskCache = "USE_DISK_CACHE"
+	// DiskCachePath is the path to where the results will be cached on disk.
```
Comment (azeemshaikh38) [inline roundtripper/roundtripper.go:40]: "cached on disk if `UseDiskCache` is true"

File: roundtripper/roundtripper.go [hunk 3]
```diff
@@ -27,15 +27,23 @@ import (
...
+	DiskCachePath = "DISK_CACHE_PATH"
+	// UseBlobCache will cache results into a blob for subsequent runs if set to true.
```
Comment (azeemshaikh38) [inline roundtripper/roundtripper.go:42]: `cache results into blob store for subsequent runs`

File: roundtripper/roundtripper.go [hunk 4]
```diff
@@ -27,15 +27,23 @@ import (
...
+	UseBlobCache = "USE_BLOB_CACHE"
+	// BucketURL is the URL where the blob cache will be stored.
```
Comment (azeemshaikh38) [inline roundtripper/roundtripper.go:44]: URL for blob store cache if `UseBlobCache`is true.

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by chrismcgehee** [comment]
Azeem, thanks for the suggestions. I've made the updates.

File: roundtripper/roundtripper.go [hunk 5]
```diff
@@ -27,15 +27,23 @@ import (
...
-	cacheSize               uint64 = 10000 * 1024 * 1024 // 10gb
+	// GithubAuthToken is for making requests to GiHub's API.
+	GithubAuthToken = "GITHUB_AUTH_TOKEN" // #nosec G101
+	// GithubAppKeyPath is the path to file for GitHub App key.
+	GithubAppKeyPath = "GITHUB_APP_KEY_PATH"
+	// GithubAppID is the app ID for the GitHub App.
+	GithubAppID = "GITHUB_APP_ID"
+	// GithubAppInstallationID is the installation ID for the GitHub App.
+	GithubAppInstallationID = "GITHUB_APP_INSTALLATION_ID"
+	// UseDiskCache will cache results into blob store for subsequent runs if set to true.
```
Comment (azeemshaikh38) [inline roundtripper/roundtripper.go:38]: Its `disk` here on this line and `blob store` belo.

Comment (chrismcgehee) [reply roundtripper/roundtripper.go]: Ah, crap. I've made these updates now.

**Review by azeemshaikh38 — APPROVED** [review]

**Pull request merged**